### PR TITLE
Make logging optional

### LINKED ISSUE #33: Remove Logger from MessageSinkRegister
We should allow developers to specify their own log function to `MessageSinkRegister` or to turn off logging.

### PR DESCRIPTION
**Pull request by lewandowski-jan**
* add logging field and log function to MessageSinkRegister

* use _log function instead of Logger.info in MessageSinkRegister

Closes #33

### CODE REVIEW

**Comment by iasiu** [comment]
I will merge this one after #41

File: packages/comms/lib/src/message_sink_register.dart [hunk 1]
```diff
@@ -1,5 +1,7 @@
 part of '../comms.dart';
 
+typedef Log = void Function(String message);
```
Comment (bartekpacia) [inline packages/comms/lib/src/message_sink_register.dart:3]:
```suggestion
typedef _LoggerCallback = void Function(String message);
```

Other packages name it like this ([1](https://github.com/leancodepl/patrol/blob/master/packages/patrol/lib/src/native/native_automator.dart#L10) [2](https://github.com/leancodepl/flutter_corelibrary/blob/master/packages/login_client/lib/src/login_client.dart#L28))

File: packages/comms/lib/src/message_sink_register.dart [hunk 2]
```diff
@@ -13,8 +15,24 @@ class MessageSinkRegister {
 
   static final MessageSinkRegister _instance = MessageSinkRegister._();
 
+  static bool logging = true;
+  static Log? log;
```
Comment (bartekpacia) [inline packages/comms/lib/src/message_sink_register.dart:19]:
```suggestion
  static _LoggerCallback? logger;
```

File: packages/comms/lib/src/message_sink_register.dart [hunk 3]
```diff
@@ -13,8 +15,24 @@ class MessageSinkRegister {
 
   static final MessageSinkRegister _instance = MessageSinkRegister._();
 
+  static bool logging = true;
```
Comment (bartekpacia) [inline packages/comms/lib/src/message_sink_register.dart:18]:
```suggestion
  static bool loggingEnabled = true;
```

File: packages/comms/lib/src/message_sink_register.dart [hunk 4]
```diff
@@ -1,5 +1,7 @@
 part of '../comms.dart';
 
+typedef _LogggerCallback = void Function(String message);
```
Comment (PiotrRogulski) [inline packages/comms/lib/src/message_sink_register.dart:3]:
```suggestion
typedef _LoggerCallback = void Function(String message);
```

File: packages/comms/lib/src/message_sink_register.dart [hunk 5]
```diff
@@ -13,8 +15,24 @@ class MessageSinkRegister {
 
   static final MessageSinkRegister _instance = MessageSinkRegister._();
 
+  static bool loggingEnabled = true;
+  static _LogggerCallback? log;
```
Comment (PiotrRogulski) [inline packages/comms/lib/src/message_sink_register.dart:19]: The field is public, so shouldn't the type be as well?

**Review by PiotrRogulski — APPROVED** [review]

**Review by PiotrRogulski — APPROVED** [review]

**Pull request merged**